src/sentry/api/endpoints/organization_trace_details.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import absolute_import
+
+from django.utils import timezone
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases import OrganizationEndpoint, OrganizationPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.event_search import get_snuba_query_args
+from sentry.models import SnubaEvent
+from sentry.api.utils import MAX_STATS_PERIOD
+from sentry.tracing.logic import transform_to_spans
+from sentry.utils import snuba
+
+
+class OrganizationTracePermission(OrganizationPermission):
+    scope_map = {
+        'GET': ['event:read', 'event:write', 'event:admin'],
+    }
+
+
+class OrganizationTraceDetailsEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationTracePermission, )
+
+    def get(self, request, organization, trace_id):
+        """
+        Fetch all the events in a trace by trace_id
+
+        :pparam string trace_id: The trace id to get transactions/events for.
+        :auth: required
+        """
+        if not features.has('organizations:events-v2', organization, actor=request.user):
+            raise ResourceDoesNotExist()
+
+        projects = self.get_projects(request, organization)
+        current_time = timezone.now()
+        trace_query = {
+            'trace': trace_id,
+            'project_id': [p.id for p in projects],
+        }
+        snuba_query = get_snuba_query_args(params=trace_query)
```

Comment:
I would discourage loading traces directly as raw snuba query from the api at this point in the development because of a few reasons:
- we are still nailing down the exact storage model, most likely it will be at least a different dataset, so going around changing all the raw_query calls will be a lot harder when we get to the right model.
- it is hard to be sure about the fields we would be filtering on at this point.

I suggest you add a simple transacitonstore abstraction that will decouple the api development from the data store specific details.